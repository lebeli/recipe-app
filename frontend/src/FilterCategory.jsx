import React from "react";
import ToggleButton from "@material-ui/lab/ToggleButton";
import "./FilterCategory.scss";

function FilterCategory(params) {
  const [selected, setSelected] = React.useState(false);

  return (
    <div className="FilterCategory">
      <ToggleButton
        value={params.value}
        selected={selected}
        disableFocusRipple
        onChange={() => {
          params.updateState(params.value, !params.selected);
          setSelected(!params.selected);
        }}
      >
        {params.categoryName}
      </ToggleButton>
    </div>
  );
}

export default FilterCategory;
