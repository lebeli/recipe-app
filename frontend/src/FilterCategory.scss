@import "./styles/styles.scss";

.FilterCategory {
  .MuiToggleButton-root {
    border-radius: 24px;
    border: 0px;
    background-color: white;
    height: 40px;
    padding: 0 40px 0 40px;

    .MuiToggleButton-label {
      font-family: $secondaryFont;
      text-transform: none;
      color: black;
    }
  }

  .MuiToggleButton-root:hover {
    border-radius: 24px;
    border: 0px;
    background-color: white;
    height: 40px;
    padding: 0 40px 0 40px;

    .MuiToggleButton-label {
      font-family: $secondaryFont;
      text-transform: none;
      color: black;
    }
  }

  @media screen and (min-width: 1024px) {
    .MuiToggleButton-root:hover {
      background-color: $primaryColor;

      .MuiToggleButton-label {
        color: white;
      }
    }
  }

  .MuiToggleButton-root.Mui-selected {
    background-color: $primaryColor;

    .MuiToggleButton-label {
      color: white;
    }
  }

  .MuiToggleButton-root.Mui-selected:hover {
    background-color: $primaryColorDarker;

    .MuiToggleButton-label {
      color: white;
    }
  }
}
