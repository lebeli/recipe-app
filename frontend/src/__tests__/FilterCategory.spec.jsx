import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import FilterCategory from "../FilterCategory";

describe("<FilterCategory />", () => {
  test("should render elements", () => {
    const wrapper = mount(
      <FilterCategory
        categoryName="Fr端hst端ck"
        value="breakfast"
        selected={false}
      />
    );
    expect(wrapper.find(".FilterCategory"));
  });

  test("renders correctly", () => {
    const tree = renderer
      .create(
        <FilterCategory
          categoryName="Fr端hst端ck"
          value="breakfast"
          selected={false}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
