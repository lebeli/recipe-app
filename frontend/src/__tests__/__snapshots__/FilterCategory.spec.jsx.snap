// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterCategory /> renders correctly 1`] = `
<div
  className="FilterCategory"
>
  <button
    aria-pressed={false}
    className="MuiButtonBase-root MuiToggleButton-root"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
    value="breakfast"
  >
    <span
      className="MuiToggleButton-label"
    >
      Frühstück
    </span>
  </button>
</div>
`;
